### ISSUE TITLE
Building coreclr tests with the Windows SDK v10.0.19025.0 Preview installed fails

### ISSUE THREAD

**Issue by tannergooding** [dotnet/runtime#908]
If you have installed the Windows SDK v10.0.19025.0 Preview and you attempt to build coreclr tests, the build for `tests\src\Interop\WinRT\NativeComponent` fails with multiple errors.

The unique errors are as follows (several of the errors are duplicated on other lines or sections of code):
```
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.UriTesting.cpp(11,19): error C3779: 'winrt::impl::consume_Windows_Foundation_IStringable<D>::ToString': a function that returns'auto' cannot be used before it is defined [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Program Files (x86)\Windows Kits\10\Include\10.0.19025.0\cppwinrt\winrt\base.h(6022,3): error C2259: 'winrt::impl::produce<D,I>': cannot instantiate abstract class [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.KeyValuePairTesting.cpp(46,104): error C2039: 'single_threaded_vector': is not a member of 'winrt' [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.KeyValuePairTesting.cpp(46): error C2259: 'pair_wrapper<int32_t,winrt::Windows::Foundation::Collections::IIterable<int32_t>>': cannot instantiate abstract class [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.NullableTesting.cpp(16,23): error C2039: 'unbox_value': is not a member of 'winrt' [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.NullableTesting.cpp(16,34): error C2065: 'unbox_value': undeclared identifier [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.NullableTesting.cpp(16,35): error C2275: 'int32_t': illegal use of this type as an expression [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
C:\Users\tagoo\source\repos\runtime\src\coreclr\tests\src\Interop\WinRT\NativeComponent\Component.Contracts.BindingViewModel.h(152,35): error C2259: 'ObservableCollection<int,std::vector<T,std::allocator<int>>>': cannot instantiate abstract class [C:\Users\tagoo\source\repos\runtime\artifacts\tests\coreclr\obj\Windows_NT.x64.Debug\Native\src\Interop\WinRT\NativeComponent\WinRTNativeComponent.vcxproj]
```

**Comment by jeffschwMSFT** [comment]
cc @jkoritzinsky

**Comment by jkoritzinsky** [comment]
Looks like the new sdk's version of cppwinrt broke compat with some of the features I was using in our WinRT tests. I'll take a look.

**Comment by AaronRobinsonMSFT** [comment]
@jkoritzinsky Any update on this? Can it be closed?

**Comment by jkoritzinsky** [comment]
I haven’t had a chance to look into fixing this. Been too busy with everything else I’m working on.

**Comment by tannergooding** [comment]
Any progress on this? Windows 20H1 will presumably be coming out in the next couple months (based on past releases) and having the preview SDK installed is still causing build failures.

**Comment by tannergooding** [comment]
(Noting the preview number is now v10.0.19041.0)

**Comment by jkotas** [comment]
Fixed by #34513
